### DISCUSSION TITLE
Hiding a Page widget from Panel Dashboard

### DISCUSSION THREAD

**Discussion by ccc-developer** [filamentphp/filament#8743] [Help]
I recently refactored by Filament application to create two panels, Admin and Dashboard. Users login and see Dashboard panel. Everything is working after the refactor, except the StatsOver widget that I created for a Page called, "Profit & Loss".

The widget is registered on the Profit & Loss page, so it loads there; however, the problem is the StatsOver widget **also** displays on the User Panel Dashboard. I've tried moving it out the default widget namespace, but throws an error, so I'm really stumped.

I've read the documentation, searched Google, watched YouTube videos, but I can't figure out how to remove the StatsOver widget from the User Panel Dashboard **and** ensure that the widget loads on the "Profit & Loss" Page. 

I've disabled the default widgets in the UserPanelProvider.php
`            ->widgets([
                //
            ])`

Any guidance is appreciated. I'm using Filament v3.0.62. Laravel 10.25, PHP 8.2.

**Comment by VincvanAs** [comment]
Could you check/share 

`->discoverWidgets(in: app_path('Filament/Widgets'), for: 'App\\Filament\\Widgets') `

on both PanelProviders?

**Reply by ccc-developer** [reply]
@VincvanAs thanks for replying. 

Here's the AdminPanelProvider.php
`->discoverResources(in: app_path('Filament/Admin/Resources'), for: 'App\\Filament\\Admin\\Resources')
->discoverPages(in: app_path('Filament/Admin/Pages'), for: 'App\\Filament\\Admin\\Pages')
->pages([
            ])
->discoverWidgets(in: app_path('Filament/Admin/Widgets'), for: 'App\\Filament\\Admin\\Widgets')
->widgets([
            ])`

Here's the UserPanelProvider.php:
`->discoverResources(in: app_path('Filament/User/Resources'), for: 'App\\Filament\\User\\Resources')
->discoverPages(in: app_path('Filament/User/Pages'), for: 'App\\Filament\\User\\Pages\\')
->pages([
Dashboard::class,
ProfitLossReport::class,
])
->discoverWidgets(in: app_path('Filament/User/Widgets'), for: 'App\\Filament\\User\\Widgets')
->widgets([
 //
 ])`

**Comment by VincvanAs** [comment]
Hmm, kinda hard to see what is going here.

You could try to override the Admin Dashboard (if you haven't already / + i don't know if this is a wanted solution for you as well).

in app/Filament/Admin /Pages create a Dashboard.php

```
<?php

namespace App\Filament\Admin\Pages;

use Filament\Pages\Dashboard as BasePage;

class Dashboard extends BasePage
{

    #protected static string $view = 'filament.admin.pages.dashboard';
    
}
```

**Reply by KLA6** [reply]
In my case, making `App\Filament\Admin\Pages\Dashboard.php` had no effect,

but making `App\Filament\\Pages\Dashboard.php` had an effect to show a new link on the left navigation.

```
<?php

namespace App\Filament\Pages;

use Filament\Pages\Dashboard as BasePage;

class Dashboard extends BasePage {

  protected static string $view = 'filament.pages.test'; # This does NOT work.

} # class
```

However, it does NOT call the view file that I've set as the above.

The view file is located at `resource\views\filament\pages\test.blade.php`.

In case of Dashboard, is there any different way to set the view file...?

**Comment by brunogdn** [comment]
You can set the isDiscovered attribute to false on your widgets and only call the specific widgets on each panel.

```
 protected static bool $isDiscovered = false;
```

**Reply by alvinChristianto** [reply]
amazing, thanks !

**Reply by Pavel-Ivanov** [reply]
Thanks!

**Reply by maxlingenfelter** [reply]
Very helpfull!

**Reply by Zxce3** [reply]
this is what i need, thanks

**Reply by eduardocm2002** [reply]
gracias

**Reply by Co-lossus** [reply]
thanks

**Comment by emmanpbarrameda** [comment]
**To Hide Widget on Dashboard,**
Open Dashboard.php located in ``vendor\filament\filament\src\Pages\Dashboard.php.``

Update the getWidgets method to filter out the widget(s) you don’t want to display.

```
    public function getWidgets(): array
    {
        // get all
        $widgets = Filament::getWidgets();
        
        // but filter out
        return array_filter($widgets, function ($widget) {
            return $widget !== \App\Filament\Widgets\WidgetClassNameHere::class;
        });
    }
